### DISCUSSION TITLE
Use embedded Calibre metadata for PDFs if available

### DISCUSSION THREAD

**Discussion by microtherion** [Kareadita/Kavita#3103] [Ideas]
### Idea Description

Currently, Kavita only uses the file name for PDF metadata. But in fact, tools such as calibre do add inline metadata when saving PDFs. Here's an example file demonstrating this, with the metadata appearing inside an `x:xmpmeta` tag.

The format seems to consist of a general RDF description element (which may in fact be written by other tools as well, although I'm not currently aware of any), as well as a calibre description element which adds some tags presumably not defined in the general element.

[Experiments in PDF Metadata.pdf](https://github.com/user-attachments/files/16567588/Experiments.in.PDF.Metadata.pdf)

### Idea Category

Feature Enhancement

### Duration of Using Kavita

A few months

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by microtherion** [comment]
The metadata written for this example is:
```
<?xpacket begin="���" id="W5M0MpCehiHzreSzNTczkc9d"?>
<x:xmpmeta xmlns:x="adobe:ns:meta/">
  <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <rdf:Description xmlns:dc="http://purl.org/dc/elements/1.1/" rdf:about="">
      <dc:title>
        <rdf:Alt>
          <rdf:li xml:lang="x-default">calibre PDF metadata</rdf:li>
        </rdf:Alt>
      </dc:title>
      <dc:description>
        <rdf:Alt>
          <rdf:li xml:lang="x-default"></rdf:li>
        </rdf:Alt>
      </dc:description>
      <dc:creator>
        <rdf:Seq>
          <rdf:li>Matt Neeracher</rdf:li>
        </rdf:Seq>
      </dc:creator>
      <dc:subject>
        <rdf:Bag>
          <rdf:li>some tag</rdf:li>
        </rdf:Bag>
      </dc:subject>
      <dc:publisher>
        <rdf:Bag>
          <rdf:li>Github Issue</rdf:li>
        </rdf:Bag>
      </dc:publisher>
      <dc:date>
        <rdf:Seq>
          <rdf:li>2024-08-16T00:00:00+02:00</rdf:li>
        </rdf:Seq>
      </dc:date>
      <dc:language>
        <rdf:Bag>
          <rdf:li>en</rdf:li>
        </rdf:Bag>
      </dc:language>
    </rdf:Description>
    <rdf:Description xmlns:xmp="http://ns.adobe.com/xap/1.0/" xmlns:xmpidq="http://ns.adobe.com/xmp/Identifier/qual/1.0/" rdf:about="">
      <xmp:MetadataDate>2024-08-10T01:00:44.022867+02:00</xmp:MetadataDate>
    </rdf:Description>
    <rdf:Description xmlns:prism="http://prismstandard.org/namespaces/basic/2.0/" rdf:about=""/>
    <rdf:Description xmlns:pdfx="http://ns.adobe.com/pdfx/1.3/" rdf:about=""/>
    <rdf:Description xmlns:calibre="http://calibre-ebook.com/xmp-namespace" xmlns:calibreSI="http://calibre-ebook.com/xmp-namespace-series-index" xmlns:calibreCC="http://calibre-ebook.com/xmp-namespace-custom-columns" rdf:about="">
      <calibre:rating>8</calibre:rating>
      <calibre:series rdf:parseType="Resource">
        <rdf:value>Experiments in PDF Metadata</rdf:value>
        <calibreSI:series_index>20.00</calibreSI:series_index>
      </calibre:series>
      <calibre:timestamp>2024-08-10T00:58:00.805850+02:00</calibre:timestamp>
      <calibre:title_sort>metadata, calibre PDF</calibre:title_sort>
      <calibre:author_sort>Neeracher, Matthias</calibre:author_sort>
    </rdf:Description>
  </rdf:RDF>
</x:xmpmeta>
```

**Comment by microtherion** [comment]
Per discord discussion, I'll take a shot at implementing this myself.

**Comment by bhcompy** [comment]
Any update on this?  Extremely interested

**Comment by microtherion** [comment]
I had a working first implementation, which was rather naive and not performant. Currently reimplementing in a more principled way, but somewhat time constrained at the moment.

**Reply by bhcompy** [reply]
Looking forward to it.  Thank you!

**Reply by 42Craft** [reply]
> I had a working first implementation, which was rather naive and not performant. Currently reimplementing in a more principled way, but somewhat time constrained at the moment.

Looking forward to it :)

**Reply by slaszt** [reply]
Have you had the time to revisit this? No pressure at all, just wondering.

**Comment by microtherion** [comment]
Good timing! I was busy with other projects and returned to it this week, and finally making progress. Learning more about the PDF spec than I ever wanted to learn :grin:

**Comment by majora2007** [comment]
This will be available in v0.8.5. Thanks again @microtherion for driving this home.
